File: 1/sol.clj
(defn get-lines [file]
  (with-open [rdr (clojure.java.io/reader file)]
    (doall (line-seq rdr))))

(def lines
  (map #(Integer/parseInt %) (get-lines "./input.txt")))


(defn queue
  ([] (clojure.lang.PersistentQueue/EMPTY))
  ([coll]
    (reduce conj clojure.lang.PersistentQueue/EMPTY coll)))

(defn solve [coll k]
  (loop [c (drop k coll)
         q (queue (take k  coll))
         s (apply + q) 
         res 0]
    (cond
      (empty? c) res 
      :else (let [ss (+ (- s (peek q)) (first c))]
              (recur (rest c)
                     (conj (pop q) (first c))
                     ss
                     (if (< s ss) (inc res) res))))))

(solve lines 1)
(solve lines 3)

File: 1/sol.py
path = "./input.txt"
with open(path, "r") as f:
    l = [int(s) for s in f.read().splitlines()]

def solve(l, k):
    s = sum(l[:k])    
    res = 0
    for i in range(k, len(l)):
        ss = s - l[i - k] + l[i]
        res += s < ss
        s = ss
    return res

print(solve(l, 1))
print(solve(l, 3))

